integration.js

```diff
@@ -143,7 +185,14 @@ IntegrationClient.prototype.request = function (method, path, payload, callback)
         callback(new Error('Received non-JSON response: ' + buff))
       }
       if (res.statusCode >= 200 && res.statusCode < 300) {
-        handleEffects(self, data, callback)
+        switch (path) {
```

Comment:
Rather than a `switch` on the request path, a nicer pattern would be wrapping the `callback` passed to methods that trigger effects.